app/src/main/java/com/alphawallet/app/service/KeyService.java

```diff
@@ -254,7 +253,7 @@ public void getAuthenticationForSignature(Wallet wallet, Activity callingActivit
 
         switch (wallet.type)
         {
-            case KEYSTORE_LEGACY:
+            case PRIVATE_KEY:
                 signCallback.GotAuthorisation(true); //Legacy keys don't require authentication
```

Comment:
@JamesSmartCell isn't this determined by whether the key is locked by the enclave rather than it's type? just because I import a private key, doesn't mean it can't be authenticated with the enclave

Reply:
No, legacy keys will never require authentication because these are legacy from before we allowed user to lock keys, and are handled by a different encryption algorithm.

Private key should never be seen here because that is only for import, once you import a private key it's converted into a KEYSTORE type for storage, so that switch statement will never trigger, however it will break signing with a key left over from the past.

A better fix would be to automatically convert all legacy keys to KEYSTORE type at startup which actually would simplify the code somewhat.

What is the actual issue you're seeing? When you import with a private key you should only be seeing that it's a yellow bar.

Reply:
@JamesSmartCell I import one of @jzaki keys and I cannot make any transactions as it hits a problem in Authentication failed 

public enum AuthenticationFailType
{
    AUTHENTICATION_DIALOG_CANCELLED,
    FINGERPRINT_NOT_VALIDATED, PIN_FAILED,
    DEVICE_NOT_SECURE
}

We hit PIN_FAILED & DEVICE_NOT_SECURE both on the physical device and emulator, hence I thought it might be related...

Reply:
This is on a device that has no pin or lock?

Reply:
I just tested this: follow this flow on a non secured device:

- import private key
- attempt to move tokens or eth to another address